### PR TITLE
Docs update for point release

### PR DESCRIPTION
**Pull request by uellue**
## Contributor Checklist:

* [x] I have added or updated my entry in [the creators.json file](https://github.com/LiberTEM/LiberTEM/blob/master/packaging/creators.json)
* [x] I have added [a changelog entry](https://github.com/LiberTEM/LiberTEM/tree/master/docs/source/changelog) for my contribution
* [x] I have added/updated documentation for all user-facing changes
* [x] I have added/updated test cases
* [x] I have included the [rebuilt production build of the client](https://libertem.github.io/LiberTEM/contributing.html?#building-the-client) (only if changes were made to the GUI)

## Reviewer Checklist:

* [ ] `/azp run libertem.libertem-data` passed
* [x] No import of GPL code from MIT code

### CODE REVIEW

File: docs/source/releasing.rst [hunk 1]
```diff
@@ -22,6 +22,16 @@ If you are bumping to a .dev0 suffix, omit :code:`--tag` and only pass :code:`--
 
     $ ./scripts/release bump v0.4.0.dev0 --commit
 
+By default, the release script will refuse to downgrade the version. This can be
+overridden with the :code:`--force` option, for example to prepare a point
+release. In that case the repository should be prepared to contain only the
```
Comment (sk1p) [inline docs/source/releasing.rst:27]: In other words, it should only be done on the master branch if no unrelated changes have been merged yet, otherwise a stable branch should be used

Comment (uellue) [reply docs/source/releasing.rst:27]: Ok, will add!

**Review by matbryan52 — APPROVED** [review]
Looks good to me, within what I understand of the release script 👍
